# Notebook for running all the examples in the examples folder

This notebook also serves as a test of the ras_commander package using the HEC-RAS Project Examples provided by HEC.

Example project names can be found in the example_projects.csv file after the examples are run.

To use a different example, change the project name in the examples and run them directly or run this notebook.  


The notebook format provides a convenient way to run all the examples to ensure that no features are broken after code revisions.  


-----

This table is a valuable reference for formulating examples and anticipating script behavior.

| Project           | Type     | Existing Numbers | Next Available |
|-------------------|----------|-------------------|----------------|
| Balde Eagle Creek | Plan     | 01, 02            | 03             |
|                   | Flow     | 01, 02            | 03             |
|                   | Unsteady | 02                | 01, 03         |
|                   | Geometry | 01                | 02             |
| Muncie            | Plan     | 01, 03, 04        | 02, 05         |
|                   | Flow     | 01                | 02             |
|                   | Unsteady | 01                | 02             |
|                   | Geometry | 01, 02, 04        | 03, 05         |

Notes:
1. For Balde Eagle Creek's Unsteady entries, 01 is available before 03. The library should use the lowest available number when creating new entries.
2. For Muncie's Plan entries, both 02 and 05 are available. The library should use 02 first.
3. For Muncie's Geometry entries, 03 is available before 05.

When writing examples or scripts:
1. Use existing numbers when referencing current plans, flows, unsteady setups, or geometries.
2. When creating new entries (e.g., cloning), use the "Next Available" number and then increment for subsequent creations.
3. After operations that create new entries, verify that the ras object has been updated to reflect these changes.

This approach will help ensure that examples run as expected and that we're always working with valid, up-to-date data in our scripts.

-----

# RAS-Commander (ras_commander) library examples

### The Examples Below Have Been Tested and Verified.

-----

In [ ]:
#!pip install pandas pathlib requests psutil h5py matplotlib scipy

In [ ]:
# Example 1: Import and execute 01_project_initialization.py
print("Executing 01_project_initialization.py:")
%run 01_project_initialization.py
print("\n")

-----

In [ ]:
import time

# wait 10 seconds
time.sleep(10)

In [ ]:
# Example 2: Import and execute 02_plan_operations.py
print("Executing 02_plan_operations.py:")
%run 02_plan_operations.py
print("\n")

-----

In [ ]:
# wait 10 seconds
time.sleep(10)

In [ ]:
# Example 3: Import and execute 03_geometry_operations.py
print("Executing 03_geometry_operations.py:")
%run 03_geometry_operations.py
print("\n")

-----

In [ ]:
# wait 10 seconds
time.sleep(10)

In [ ]:
# Example 4: Import and execute 04_unsteady_flow_operations.py
print("Executing 04_unsteady_flow_operations.py:")
%run 04_unsteady_flow_operations.py
print("\n")

-----

In [ ]:
# wait 10 seconds
time.sleep(10)

In [ ]:
# Example 5: Import and execute 05_utility_functions.py
print("Executing 05_utility_functions.py:")
%run 05_utility_functions.py
print("\n")

-----

In [ ]:
# wait 10 seconds
time.sleep(10)

In [ ]:
# Example 6: Import and execute 06_single_plan_execution.py
print("Executing 06_single_plan_execution.py:")
%run 06_single_plan_execution.py
print("\n")

-----

In [ ]:
# wait 10 seconds
time.sleep(10)

In [ ]:
# Example 7: Import and execute 07_sequential_plan_execution.py
print("Executing 07_sequential_plan_execution.py:")
%run 07_sequential_plan_execution.py
print("\n")

-----

In [ ]:
# wait 10 seconds
time.sleep(10)

In [ ]:
# Example 8: Import and execute 08_parallel_execution.py
print("Executing 08_parallel_execution.py:")
%run 08_parallel_execution.py
print("\n")

-----

In [ ]:
# wait 10 seconds
time.sleep(10)

In [ ]:
# Example 9: Import and execute 09_specifying_plans.py
print("Executing 09_specifying_plans.py:")
%run 09_specifying_plans.py
print("\n")

-----

In [ ]:
# wait 10 seconds
time.sleep(10)

In [4]:
print("Executing 10_arguments_for_compute.py:")
%run 10_arguments_for_compute.py
print("\n")

2024-10-07 18:25:56,595 - ras_commander.logging_config - DEBUG - Calling __init__
2024-10-07 18:25:56,596 - ras_commander.RasExamples - INFO - Example projects folder: c:\GH\ras-commander\examples\example_projects
2024-10-07 18:25:56,597 - ras_commander.logging_config - DEBUG - Calling _load_project_data
2024-10-07 18:25:56,597 - ras_commander.logging_config - DEBUG - Calling _find_zip_file
2024-10-07 18:25:56,599 - ras_commander.RasExamples - INFO - Found zip file: c:\GH\ras-commander\examples\Example_Projects_6_5.zip
2024-10-07 18:25:56,600 - ras_commander.logging_config - DEBUG - Finished _find_zip_file
2024-10-07 18:25:56,602 - ras_commander.RasExamples - INFO - Loading project data from CSV...
2024-10-07 18:25:56,605 - ras_commander.RasExamples - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-10-07 18:25:56,605 - ras_commander.logging_config - DEBUG - Finished _load_project_data
2024-10-07 18:25:56,606 - ras_commander.logging_co

Executing 10_arguments_for_compute.py:
Available plans:
  plan_number                                          full_path  \
0          01  c:\GH\ras-commander\examples\example_projects\...   
1          02  c:\GH\ras-commander\examples\example_projects\...   

  Computation Interval DSS File Flow File Friction Slope Method Geom File  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  Mapping Interval                     Plan Title Program Version Run HTab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  Run PostProcess Run Sediment Run UNet Run WQNet Short Identifier  \
0               1            0        1         0     UnsteadyFlow   
1               1          NaN        1       NaN        SteadyRun   

                   Simulation Date UNET D1 Cores UNET Use Exi

2024-10-07 18:25:56,780 - ras_commander.logging_config - DEBUG - Calling initialize
2024-10-07 18:25:56,781 - ras_commander.logging_config - DEBUG - Calling find_ras_prj
2024-10-07 18:25:56,784 - ras_commander.logging_config - DEBUG - Finished find_ras_prj
2024-10-07 18:25:56,785 - ras_commander.logging_config - DEBUG - Calling _load_project_data
2024-10-07 18:25:56,785 - ras_commander.logging_config - DEBUG - Calling _get_prj_entries
2024-10-07 18:25:56,787 - ras_commander.logging_config - DEBUG - Calling _parse_plan_file
2024-10-07 18:25:56,788 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek _01_[SequentialWithArgs]\BaldEagle.p01
2024-10-07 18:25:56,789 - ras_commander.logging_config - DEBUG - Finished _parse_plan_file
2024-10-07 18:25:56,789 - ras_commander.logging_config - DEBUG - Calling _parse_plan_file
2024-10-07 18:25:56,791 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_

Plan 01 execution: Successful


2024-10-07 18:28:12,468 - ras_commander.logging_config - DEBUG - Finished get_geom_entries
2024-10-07 18:28:12,468 - ras_commander.logging_config - DEBUG - Calling get_flow_entries
2024-10-07 18:28:12,469 - ras_commander.logging_config - DEBUG - Calling check_initialized
2024-10-07 18:28:12,470 - ras_commander.logging_config - DEBUG - Finished check_initialized
2024-10-07 18:28:12,470 - ras_commander.logging_config - DEBUG - Calling _get_prj_entries
2024-10-07 18:28:12,473 - ras_commander.logging_config - DEBUG - Finished _get_prj_entries
2024-10-07 18:28:12,473 - ras_commander.logging_config - DEBUG - Finished get_flow_entries
2024-10-07 18:28:12,474 - ras_commander.logging_config - DEBUG - Calling get_unsteady_entries
2024-10-07 18:28:12,475 - ras_commander.logging_config - DEBUG - Calling check_initialized
2024-10-07 18:28:12,476 - ras_commander.logging_config - DEBUG - Finished check_initialized
2024-10-07 18:28:12,476 - ras_commander.logging_config - DEBUG - Calling _get_prj_entri

Plan 02 execution: Successful
Sequential execution completed

Example 2: Parallel execution with various arguments


2024-10-07 18:28:20,603 - ras_commander.RasCmdr - INFO - Created worker folder: c:\GH\ras-commander\examples\example_projects\parallel_results [Worker 2]
2024-10-07 18:28:20,604 - ras_commander.logging_config - DEBUG - Calling init_ras_project
2024-10-07 18:28:20,605 - ras_commander.logging_config - DEBUG - Calling get_ras_exe
2024-10-07 18:28:20,607 - ras_commander.RasPrj - INFO - HEC-RAS executable found at specified path: C:\Program Files (x86)\HEC\HEC-RAS\6.6\Ras.exe
2024-10-07 18:28:20,608 - ras_commander.logging_config - DEBUG - Finished get_ras_exe
2024-10-07 18:28:20,609 - ras_commander.logging_config - DEBUG - Calling initialize
2024-10-07 18:28:20,610 - ras_commander.logging_config - DEBUG - Calling find_ras_prj
2024-10-07 18:28:20,614 - ras_commander.logging_config - DEBUG - Finished find_ras_prj
2024-10-07 18:28:20,615 - ras_commander.logging_config - DEBUG - Calling _load_project_data
2024-10-07 18:28:20,616 - ras_commander.logging_config - DEBUG - Calling _get_prj_entries

Parallel execution results:
Plan 01: Successful
Plan 02: Successful

Example 3: Single plan execution with specific arguments


2024-10-07 18:30:44,904 - ras_commander.logging_config - DEBUG - Calling set_num_cores
2024-10-07 18:30:44,906 - ras_commander.logging_config - DEBUG - Calling check_initialized
2024-10-07 18:30:44,907 - ras_commander.logging_config - DEBUG - Finished check_initialized
2024-10-07 18:30:44,908 - ras_commander.logging_config - DEBUG - Calling get_plan_path
2024-10-07 18:30:44,909 - ras_commander.logging_config - DEBUG - Calling check_initialized
2024-10-07 18:30:44,910 - ras_commander.logging_config - DEBUG - Finished check_initialized
2024-10-07 18:30:44,912 - ras_commander.RasUtils - INFO - Using provided plan file path: c:\GH\ras-commander\examples\example_projects\compute_test_2\BaldEagle.p02
2024-10-07 18:30:44,913 - ras_commander.logging_config - DEBUG - Finished get_plan_path
2024-10-07 18:30:44,914 - ras_commander.logging_config - DEBUG - Calling update_file
2024-10-07 18:30:44,920 - ras_commander.RasUtils - INFO - Successfully updated file: c:\GH\ras-commander\examples\example_p

Single plan execution: Successful




In [ ]:
# wait 10 seconds
time.sleep(10)

In [ ]:
print("Executing 12_plan_set_execution.py:")
%run 12_plan_set_execution.py
print("\n")

In [ ]:
# wait 10 seconds
time.sleep(10)

In [3]:
print("Executing 13_multiple_project_operations.py:")
%run 13_multiple_project_operations.py
print("\n")

2024-10-07 17:58:57,043 - ras_commander.logging_config - DEBUG - Calling __init__
2024-10-07 17:58:57,044 - ras_commander.RasExamples - INFO - Example projects folder: c:\GH\ras-commander\examples\example_projects
2024-10-07 17:58:57,044 - ras_commander.logging_config - DEBUG - Calling _load_project_data
2024-10-07 17:58:57,045 - ras_commander.logging_config - DEBUG - Calling _find_zip_file
2024-10-07 17:58:57,046 - ras_commander.RasExamples - INFO - Found zip file: c:\GH\ras-commander\examples\Example_Projects_6_5.zip
2024-10-07 17:58:57,047 - ras_commander.logging_config - DEBUG - Finished _find_zip_file
2024-10-07 17:58:57,048 - ras_commander.RasExamples - INFO - Loading project data from CSV...
2024-10-07 17:58:57,050 - ras_commander.RasExamples - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-10-07 17:58:57,051 - ras_commander.logging_config - DEBUG - Finished _load_project_data
2024-10-07 17:58:57,052 - ras_commander.logging_co

Executing 13_multiple_project_operations.py:


2024-10-07 17:58:57,488 - ras_commander.RasExamples - INFO - Successfully extracted project 'Muncie' to c:\GH\ras-commander\examples\example_projects\Muncie
2024-10-07 17:58:57,488 - ras_commander.RasExamples - INFO - ----- RasExamples Extraction Complete -----
2024-10-07 17:58:57,489 - ras_commander.logging_config - DEBUG - Finished extract_project
2024-10-07 17:58:57,490 - ras_commander.logging_config - DEBUG - Calling init_ras_project
2024-10-07 17:58:57,491 - ras_commander.logging_config - DEBUG - Calling get_ras_exe
2024-10-07 17:58:57,493 - ras_commander.RasPrj - INFO - HEC-RAS executable found at default path: C:\Program Files (x86)\HEC\HEC-RAS\6.6\Ras.exe
2024-10-07 17:58:57,493 - ras_commander.logging_config - DEBUG - Finished get_ras_exe
2024-10-07 17:58:57,494 - ras_commander.RasPrj - INFO - Initializing global 'ras' object via init_ras_project function.
2024-10-07 17:58:57,494 - ras_commander.logging_config - DEBUG - Calling initialize
2024-10-07 17:58:57,494 - ras_commande

Available plans in Bald Eagle Creek project:
  plan_number                                          full_path  \
0          01  c:\GH\ras-commander\examples\example_projects\...   
1          03  c:\GH\ras-commander\examples\example_projects\...   
2          04  c:\GH\ras-commander\examples\example_projects\...   

  Computation Interval DSS File Flow File Friction Slope Method Geom File  \
0                15SEC      dss       u01                     1       g01   
1                10SEC      dss       u01                     1       g02   
2                10SEC      dss       u01                     1       g04   

  Mapping Interval                                Plan Title Program Version  \
0             5MIN                  Unsteady Multi  9-SA run            5.00   
1             5MIN            Unsteady Run with 2D 50ft Grid            5.10   
2             5MIN  Unsteady Run with 2D 50ft User n Value R            5.10   

   ... Run UNet Run WQNet     Short Identifier      

2024-10-07 17:58:57,746 - ras_commander.logging_config - DEBUG - Calling _parse_plan_file
2024-10-07 17:58:57,747 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_projects\Muncie\Muncie.p03
2024-10-07 17:58:57,748 - ras_commander.logging_config - DEBUG - Finished _parse_plan_file
2024-10-07 17:58:57,748 - ras_commander.logging_config - DEBUG - Calling _parse_plan_file
2024-10-07 17:58:57,749 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_projects\Muncie\Muncie.p04
2024-10-07 17:58:57,750 - ras_commander.logging_config - DEBUG - Finished _parse_plan_file
2024-10-07 17:58:57,750 - ras_commander.logging_config - DEBUG - Calling _parse_plan_file
2024-10-07 17:58:57,751 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_projects\Muncie\Muncie.p02
2024-10-07 17:58:57,751 - ras_commander.logging_config - DEBUG - Finished _parse_plan_file
2024-10-07 17:58:57,752 - ras_commander.

Set geometry for plan 02 in Bald Eagle Creek project
Set geometry for plan 05 in Muncie project

Example 3: Updating unsteady flow parameters
Updated unsteady flow parameters for both projects

Example 4: Executing plans for both projects simultaneously in separate compute folders


2024-10-07 17:58:58,271 - ras_commander.RasCmdr - INFO - Copied project folder to destination: c:\GH\ras-commander\examples\example_projects\compute_muncie
2024-10-07 17:58:58,272 - ras_commander.RasCmdr - INFO - Copied project folder to destination: c:\GH\ras-commander\examples\example_projects\compute_bald_eagle
2024-10-07 17:58:58,272 - ras_commander.logging_config - DEBUG - Calling initialize
2024-10-07 17:58:58,273 - ras_commander.logging_config - DEBUG - Calling initialize
2024-10-07 17:58:58,274 - ras_commander.logging_config - DEBUG - Calling find_ras_prj
2024-10-07 17:58:58,275 - ras_commander.logging_config - DEBUG - Calling find_ras_prj
2024-10-07 17:58:58,278 - ras_commander.logging_config - DEBUG - Finished find_ras_prj
2024-10-07 17:58:58,279 - ras_commander.logging_config - DEBUG - Calling _load_project_data
2024-10-07 17:58:58,279 - ras_commander.logging_config - DEBUG - Finished find_ras_prj
2024-10-07 17:58:58,279 - ras_commander.logging_config - DEBUG - Calling _get_

Execution results:
Plan 02 execution: Successful
Plan 05 execution: Successful

Example 5: Getting results paths
No results found for Bald Eagle Creek plan 02
No results found for Muncie plan 05

Note: The original project folders can now be edited while the compute operations are running in separate folders.




In [ ]:
# wait 10 seconds
time.sleep(10)

In [ ]:
# Example 14 is long, and is a notebook.

In [ ]:
print("Executing 15_plan_key_operations.py:")
%run 15_plan_key_operations.py
print("\n")

## EXAMPLE IS BROKEN - NEED TO FIX FUNCTION THAT HANDLES KEY UPDATES FOR PLAN FILES, ADD ONE FOR UNSTEADY FILES + EXAMPLES FOR ALL 
## FOR ALL SUPPORTED PLAN AND UNSTEADY KEYS


In [ ]:
# wait 10 seconds
time.sleep(10)

In [1]:
print("Executing 16_scanning_ras_project_info.py:")
%run 16_scanning_ras_project_info.py
print("\n")


Executing 16_scanning_ras_project_info.py:


2024-10-07 17:58:23,522 - ras_commander.logging_config - DEBUG - Calling __init__
2024-10-07 17:58:23,523 - ras_commander.RasExamples - INFO - Example projects folder: c:\GH\ras-commander\examples\example_projects
2024-10-07 17:58:23,523 - ras_commander.logging_config - DEBUG - Calling _load_project_data
2024-10-07 17:58:23,524 - ras_commander.logging_config - DEBUG - Calling _find_zip_file
2024-10-07 17:58:23,525 - ras_commander.RasExamples - INFO - Found zip file: c:\GH\ras-commander\examples\Example_Projects_6_5.zip
2024-10-07 17:58:23,526 - ras_commander.logging_config - DEBUG - Finished _find_zip_file
2024-10-07 17:58:23,527 - ras_commander.RasExamples - INFO - Loading project data from CSV...
2024-10-07 17:58:23,530 - ras_commander.RasExamples - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-10-07 17:58:23,531 - ras_commander.logging_config - DEBUG - Finished _load_project_data
2024-10-07 17:58:23,531 - ras_commander.logging_co

Summary for category '1D Unsteady Flow Hydraulics' has been written to: c:\GH\ras-commander\examples\ras_example_categories\ras-commander 1D Unsteady Flow Hydraulics summary.txt


2024-10-07 17:58:30,955 - ras_commander.RasExamples - INFO - Successfully extracted project 'BaldEagleCrkMulti2D' to c:\GH\ras-commander\examples\example_projects\BaldEagleCrkMulti2D
2024-10-07 17:58:30,956 - ras_commander.RasExamples - INFO - ----- RasExamples Extraction Complete -----
2024-10-07 17:58:30,957 - ras_commander.RasExamples - INFO - ----- RasExamples Extracting Project -----
2024-10-07 17:58:30,957 - ras_commander.RasExamples - INFO - Extracting project 'Muncie'
2024-10-07 17:58:30,958 - ras_commander.RasExamples - INFO - Project 'Muncie' already exists. Deleting existing folder...
2024-10-07 17:58:31,016 - ras_commander.RasExamples - INFO - Existing folder for project 'Muncie' has been deleted.
2024-10-07 17:58:31,511 - ras_commander.RasExamples - INFO - Successfully extracted project 'Muncie' to c:\GH\ras-commander\examples\example_projects\Muncie
2024-10-07 17:58:31,512 - ras_commander.RasExamples - INFO - ----- RasExamples Extraction Complete -----
2024-10-07 17:58:31

Summary for category '2D Unsteady Flow Hydraulics' has been written to: c:\GH\ras-commander\examples\ras_example_categories\ras-commander 2D Unsteady Flow Hydraulics summary.txt
All category summaries have been generated.
Cleaned up original extracted example projects.




In [ ]:
# wait 10 seconds
time.sleep(10)

In [2]:
print("Executing xx_edge_cases.py:")
%run xx_edge_cases.py
print("\n")

2024-10-07 17:58:40,046 - ras_commander.logging_config - DEBUG - Calling __init__
2024-10-07 17:58:40,048 - ras_commander.RasExamples - INFO - Example projects folder: c:\GH\ras-commander\examples\example_projects
2024-10-07 17:58:40,049 - ras_commander.logging_config - DEBUG - Calling _load_project_data
2024-10-07 17:58:40,050 - ras_commander.logging_config - DEBUG - Calling _find_zip_file
2024-10-07 17:58:40,051 - ras_commander.RasExamples - INFO - Found zip file: c:\GH\ras-commander\examples\Example_Projects_6_5.zip
2024-10-07 17:58:40,051 - ras_commander.logging_config - DEBUG - Finished _find_zip_file
2024-10-07 17:58:40,052 - ras_commander.RasExamples - INFO - Loading project data from CSV...
2024-10-07 17:58:40,056 - ras_commander.RasExamples - INFO - Loaded 66 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-10-07 17:58:40,056 - ras_commander.logging_config - DEBUG - Finished _load_project_data
2024-10-07 17:58:40,057 - ras_commander.logging_co

Executing xx_edge_cases.py:
Available plans:
  plan_number                                          full_path  \
0          01  c:\GH\ras-commander\examples\example_projects\...   
1          02  c:\GH\ras-commander\examples\example_projects\...   

  Computation Interval DSS File Flow File Friction Slope Method Geom File  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  Mapping Interval                     Plan Title Program Version Run HTab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  Run PostProcess Run Sediment Run UNet Run WQNet Short Identifier  \
0               1            0        1         0     UnsteadyFlow   
1               1          NaN        1       NaN        SteadyRun   

                   Simulation Date UNET D1 Cores UNET Use Existing IB Ta

2024-10-07 17:58:40,249 - ras_commander.RasCmdr - INFO - Copied project folder to compute folder: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek [SinglePlanTest]
2024-10-07 17:58:40,251 - ras_commander.logging_config - DEBUG - Calling initialize
2024-10-07 17:58:40,251 - ras_commander.logging_config - DEBUG - Calling find_ras_prj
2024-10-07 17:58:40,254 - ras_commander.logging_config - DEBUG - Finished find_ras_prj
2024-10-07 17:58:40,254 - ras_commander.logging_config - DEBUG - Calling _load_project_data
2024-10-07 17:58:40,255 - ras_commander.logging_config - DEBUG - Calling _get_prj_entries
2024-10-07 17:58:40,256 - ras_commander.logging_config - DEBUG - Calling _parse_plan_file
2024-10-07 17:58:40,257 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek [SinglePlanTest]\BaldEagle.p01
2024-10-07 17:58:40,258 - ras_commander.logging_config - DEBUG - Finished _parse_plan_file
2024-10-07 17:58:40,259 - ras_c

Execution of plan 01 completed using compute_test_mode

Example 2: Executing a single plan using compute_parallel


2024-10-07 17:58:40,778 - ras_commander.logging_config - DEBUG - Finished get_plan_entries
2024-10-07 17:58:40,778 - ras_commander.logging_config - DEBUG - Calling get_geom_entries
2024-10-07 17:58:40,779 - ras_commander.logging_config - DEBUG - Calling check_initialized
2024-10-07 17:58:40,781 - ras_commander.logging_config - DEBUG - Finished check_initialized
2024-10-07 17:58:40,781 - ras_commander.logging_config - DEBUG - Calling _get_prj_entries
2024-10-07 17:58:40,784 - ras_commander.logging_config - DEBUG - Finished _get_prj_entries
2024-10-07 17:58:40,785 - ras_commander.logging_config - DEBUG - Finished get_geom_entries
2024-10-07 17:58:40,786 - ras_commander.logging_config - DEBUG - Calling get_flow_entries
2024-10-07 17:58:40,786 - ras_commander.logging_config - DEBUG - Calling check_initialized
2024-10-07 17:58:40,787 - ras_commander.logging_config - DEBUG - Finished check_initialized
2024-10-07 17:58:40,788 - ras_commander.logging_config - DEBUG - Calling _get_prj_entries
2

Parallel execution of single plan results:
Plan 01: Failed

Example 3: Executing a single plan using compute_test_mode with a string input


2024-10-07 17:58:41,262 - ras_commander.logging_config - DEBUG - Finished _parse_plan_file
2024-10-07 17:58:41,264 - ras_commander.logging_config - DEBUG - Calling _parse_plan_file
2024-10-07 17:58:41,266 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek [SinglePlanTestString]\BaldEagle.p02
2024-10-07 17:58:41,267 - ras_commander.logging_config - DEBUG - Finished _parse_plan_file
2024-10-07 17:58:41,270 - ras_commander.logging_config - DEBUG - Finished _get_prj_entries
2024-10-07 17:58:41,271 - ras_commander.logging_config - DEBUG - Finished get_plan_entries
2024-10-07 17:58:41,272 - ras_commander.logging_config - DEBUG - Calling get_geom_entries
2024-10-07 17:58:41,274 - ras_commander.logging_config - DEBUG - Calling check_initialized
2024-10-07 17:58:41,274 - ras_commander.logging_config - DEBUG - Finished check_initialized
2024-10-07 17:58:41,275 - ras_commander.logging_config - DEBUG - Calling _get_prj_entries
2024-10

Execution of plan 02 completed using compute_test_mode with string input

Example 4: Executing a single plan using compute_parallel with a string input


2024-10-07 17:58:41,731 - ras_commander.logging_config - DEBUG - Finished _get_prj_entries
2024-10-07 17:58:41,732 - ras_commander.logging_config - DEBUG - Finished get_plan_entries
2024-10-07 17:58:41,733 - ras_commander.logging_config - DEBUG - Calling get_geom_entries
2024-10-07 17:58:41,734 - ras_commander.logging_config - DEBUG - Calling check_initialized
2024-10-07 17:58:41,735 - ras_commander.logging_config - DEBUG - Finished check_initialized
2024-10-07 17:58:41,736 - ras_commander.logging_config - DEBUG - Calling _get_prj_entries
2024-10-07 17:58:41,738 - ras_commander.logging_config - DEBUG - Finished _get_prj_entries
2024-10-07 17:58:41,739 - ras_commander.logging_config - DEBUG - Finished get_geom_entries
2024-10-07 17:58:41,740 - ras_commander.logging_config - DEBUG - Calling get_flow_entries
2024-10-07 17:58:41,741 - ras_commander.logging_config - DEBUG - Calling check_initialized
2024-10-07 17:58:41,742 - ras_commander.logging_config - DEBUG - Finished check_initialized


Parallel execution of single plan (string input) results:
Plan 01: Failed

Example 5: Attempting to execute with an empty plan list


2024-10-07 17:58:42,336 - ras_commander.RasCmdr - ERROR - Error running plan: 01
2024-10-07 17:58:42,337 - ras_commander.RasCmdr - ERROR - Error message: 
2024-10-07 17:58:42,338 - ras_commander.RasCmdr - INFO - Total run time for plan 01: 0.17 seconds
2024-10-07 17:58:42,339 - ras_commander.logging_config - DEBUG - Calling get_plan_entries
2024-10-07 17:58:42,340 - ras_commander.logging_config - DEBUG - Calling check_initialized
2024-10-07 17:58:42,340 - ras_commander.logging_config - DEBUG - Finished check_initialized
2024-10-07 17:58:42,341 - ras_commander.logging_config - DEBUG - Calling _get_prj_entries
2024-10-07 17:58:42,342 - ras_commander.logging_config - DEBUG - Calling _parse_plan_file
2024-10-07 17:58:42,345 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek [EmptyPlanList]\BaldEagle.p01
2024-10-07 17:58:42,346 - ras_commander.logging_config - DEBUG - Finished _parse_plan_file
2024-10-07 17:58:42,347 - ras_comm


Example 6: Attempting to execute with a non-existent plan number





# Guide for Locating Relevant Examples:

## Category: 1D Unsteady Flow Hydraulics

### Flow Hydrograph

| Example | Plan File |
|---------|-----------|
| Balde Eagle Creek | BaldEagle.p01, BaldEagle.p02 |
| ContractionExpansionMinorLosses | ContExp.p01, ContExp.p02, ContExp.p03, ContExp.p04 |
| Culvert Hydraulics | Beav_Culvert.p01 |
| Culverts with Flap Gates | CulvertFlap.p01 |
| Dam Breaching | BaldEagleDamBrk.p07, BaldEagleDamBrk.p02, BaldEagleDamBrk.p10, BaldEagleDamBrk.p06, BaldEagleDamBrk.p08, BaldEagleDamBrk.p04, BaldEagleDamBrk.p03, BaldEagleDamBrk.p05, BaldEagleDamBrk.p09, BaldEagleDamBrk.p11 |
| Elevation Controled Gates | ElevControledGates.p01 |
| Inline Structure with Gated Spillways | Inline_3Gates.p01 |
| Internal Stage and Flow Boundary Condition | IBStageFlowTest.p01 |
| JunctionHydraulics | JunctionHydraulics.p01, JunctionHydraulics.p02, JunctionHydraulics.p04, JunctionHydraulics.p03 |
| Lateral Strcuture with Gates | TwoGateGrps.p01 |
| Lateral Structure connected to a River Reach | 3Reach_with_Lat.p01 |
| Lateral Structure Overflow Weir | LatMultXS.p01 |
| Lateral Structure with Culverts | LatWeirCulverts.p01 |
| Lateral Structure with Culverts and Gates | Beav_Lateral.p01 |
| Levee Breaching | LeveeBreach.p01 |
| Multiple Reaches with Hydraulic Structures | 3ReachUnsteady.p01, 3ReachUnsteady.p02 |
| Pumping Station | Pumps.p01 |
| Pumping Station with Rules | PumpRule.p03 |
| Rule Operations | Canal.p01 |
| Simplified Physical Breaching | BaldEDmbrk.p01 |
| UngagedAreaInflows | Ungaged_area.p03, Ungaged_area.p02 |
| Unsteady Flow Encroachment Analysis | UnsteadyEncroachment.p03, UnsteadyEncroachment.p02 |

### Normal Depth

| Example | Plan File |
|---------|-----------|
| Balde Eagle Creek | BaldEagle.p01, BaldEagle.p02 |
| ContractionExpansionMinorLosses | ContExp.p01, ContExp.p02, ContExp.p03, ContExp.p04 |
| Culvert Hydraulics | Beav_Culvert.p01 |
| Dam Breaching | BaldEagleDamBrk.p07, BaldEagleDamBrk.p02, BaldEagleDamBrk.p10, BaldEagleDamBrk.p06, BaldEagleDamBrk.p08, BaldEagleDamBrk.p04, BaldEagleDamBrk.p03, BaldEagleDamBrk.p05, BaldEagleDamBrk.p09, BaldEagleDamBrk.p11 |
| JunctionHydraulics | JunctionHydraulics.p01, JunctionHydraulics.p02, JunctionHydraulics.p04, JunctionHydraulics.p03 |
| Lateral Structure connected to a River Reach | 3Reach_with_Lat.p01 |
| Multiple Reaches with Hydraulic Structures | 3ReachUnsteady.p01, 3ReachUnsteady.p02 |
| Rule Operations | Canal.p01 |
| Simplified Physical Breaching | BaldEDmbrk.p01 |
| UngagedAreaInflows | Ungaged_area.p03, Ungaged_area.p02 |

### Gate Opening

| Example | Plan File |
|---------|-----------|
| Balde Eagle Creek | BaldEagle.p01, BaldEagle.p02 |
| ContractionExpansionMinorLosses | ContExp.p01, ContExp.p02, ContExp.p03, ContExp.p04 |
| Dam Breaching | BaldEagleDamBrk.p07, BaldEagleDamBrk.p02, BaldEagleDamBrk.p10, BaldEagleDamBrk.p06, BaldEagleDamBrk.p08, BaldEagleDamBrk.p04, BaldEagleDamBrk.p03, BaldEagleDamBrk.p05, BaldEagleDamBrk.p09, BaldEagleDamBrk.p11 |
| Elevation Controled Gates | ElevControledGates.p01 |
| Inline Structure with Gated Spillways | Inline_3Gates.p01 |
| Lateral Strcuture with Gates | TwoGateGrps.p01 |
| Lateral Structure with Culverts and Gates | Beav_Lateral.p01 |
| Simplified Physical Breaching | BaldEDmbrk.p01 |

### Stage Hydrograph

| Example | Plan File |
|---------|-----------|
| Internal Stage and Flow Boundary Condition | IBStageFlowTest.p01 |
| Pumping Station with Rules | PumpRule.p03 |
| Rule Operations | Canal.p01 |

### Lateral Inflow Hydrograph

| Example | Plan File |
|---------|-----------|
| Dam Breaching | BaldEagleDamBrk.p07, BaldEagleDamBrk.p02, BaldEagleDamBrk.p10, BaldEagleDamBrk.p06, BaldEagleDamBrk.p08, BaldEagleDamBrk.p04, BaldEagleDamBrk.p03, BaldEagleDamBrk.p05, BaldEagleDamBrk.p09, BaldEagleDamBrk.p11 |
| Multiple Reaches with Hydraulic Structures | 3ReachUnsteady.p01, 3ReachUnsteady.p02 |
| Pumping Station | Pumps.p01 |
| Rule Operations | Canal.p01 |
| UngagedAreaInflows | Ungaged_area.p03, Ungaged_area.p02 |

### Uniform Lateral Inflow Hydrograph

| Example | Plan File |
|---------|-----------|
| Dam Breaching | BaldEagleDamBrk.p07, BaldEagleDamBrk.p02, BaldEagleDamBrk.p10, BaldEagleDamBrk.p06, BaldEagleDamBrk.p08, BaldEagleDamBrk.p04, BaldEagleDamBrk.p03, BaldEagleDamBrk.p05, BaldEagleDamBrk.p09, BaldEagleDamBrk.p11 |
| UngagedAreaInflows | Ungaged_area.p03, Ungaged_area.p02 |

### Unknown or Special Conditions

| Example | Plan File |
|---------|-----------|
| Culverts with Flap Gates | CulvertFlap.p01 |
| Elevation Controled Gates | ElevControledGates.p01 |
| Internal Stage and Flow Boundary Condition | IBStageFlowTest.p01 |
| Lateral Structure Overflow Weir | LatMultXS.p01 |
| Lateral Structure with Culverts | LatWeirCulverts.p01 |
| Levee Breaching | LeveeBreach.p01 |
| Pumping Station with Rules | PumpRule.p03 |
| UngagedAreaInflows | Ungaged_area.p03, Ungaged_area.p02 |
| Unsteady Flow Encroachment Analysis | UnsteadyEncroachment.p03, UnsteadyEncroachment.p02 |

## Category: 2D Unsteady Flow Hydraulics

# HEC-RAS Unsteady Flow Examples Analysis

## Version 6.30

### Example: Muncie

| Boundary Condition Types | Plan Files |
|--------------------------|------------|
| Flow Hydrograph          | Muncie.p01 (Unsteady Multi  9-SA run) |
| Normal Depth             | Muncie.p03 (Unsteady Run with 2D 50ft Grid) |
|                          | Muncie.p04 (Unsteady Run with 2D 50ft User n Value R) |

| Project | Version | Boundary Conditions | Plans |
|---------|---------|---------------------|-------|
| Muncie  | 6.30    | - Flow Hydrograph (White River, Muncie, Station 15696.24) | 1. Muncie.p01: Unsteady Multi 9-SA run |
|         |         | - Normal Depth (White River, Muncie, Station 237.6455)    | 2. Muncie.p03: Unsteady Run with 2D 50ft Grid |
|         |         |                                                           | 3. Muncie.p04: Unsteady Run with 2D 50ft User n Value R |

## Version 6.00

### Example: BaldEagleCrkMulti2D

| Boundary Condition Types | Plan Files |
|--------------------------|------------|
| Flow Hydrograph          | BaldEagleDamBrk.p06 (Gridded Precip - Infiltration) |
| Gate Opening             |  |
| Lateral Inflow Hydrograph |  |
| Uniform Lateral Inflow Hydrograph |  |
| Normal Depth             |  |

| Project | Version | Boundary Conditions | Plans |
|---------|---------|---------------------|-------|
| BaldEagleCrkMulti2D | 6.00 | - Flow Hydrograph | 1. BaldEagleDamBrk.p06: Gridded Precip - Infiltration |
|                     |      | - Gate Opening |  |
|                     |      | - Lateral Inflow Hydrograph |  |
|                     |      | - Uniform Lateral Inflow Hydrograph |  |
|                     |      | - Normal Depth |  |

## Version 5.10

### Example: BaldEagleCrkMulti2D

| Boundary Condition Types | Plan Files |
|--------------------------|------------|
| Flow Hydrograph          | BaldEagleDamBrk.p13 (PMF with Multi 2D Areas) |
| Gate Opening             | BaldEagleDamBrk.p15 (1d-2D Dambreak Refined Grid) |
| Lateral Inflow Hydrograph | BaldEagleDamBrk.p02 (SA to Detailed 2D Breach) |
| Uniform Lateral Inflow Hydrograph | BaldEagleDamBrk.p05 (Single 2D area with Bridges FEQ) |
| Normal Depth             | BaldEagleDamBrk.p01 (SA to Detailed 2D Breach FEQ) |

| Project | Version | Boundary Conditions | Plans |
|---------|---------|---------------------|-------|
| BaldEagleCrkMulti2D | 5.10 | - Flow Hydrograph | 1. BaldEagleDamBrk.p13: PMF with Multi 2D Areas |
|                     |      | - Gate Opening | 2. BaldEagleDamBrk.p15: 1d-2D Dambreak Refined Grid |
|                     |      | - Lateral Inflow Hydrograph | 3. BaldEagleDamBrk.p02: SA to Detailed 2D Breach |
|                     |      | - Uniform Lateral Inflow Hydrograph | 4. BaldEagleDamBrk.p05: Single 2D area with Bridges FEQ |
|                     |      | - Normal Depth | 5. BaldEagleDamBrk.p01: SA to Detailed 2D Breach FEQ |

## Version 5.00

### Example: BaldEagleCrkMulti2D

| Boundary Condition Types | Plan Files |
|--------------------------|------------|
| Flow Hydrograph          | BaldEagleDamBrk.p17 (2D to 1D No Dam) |
| Gate Opening             | BaldEagleDamBrk.p18 (2D to 2D Run) |
| Lateral Inflow Hydrograph | BaldEagleDamBrk.p19 (SA to 2D Dam Break Run) |
| Uniform Lateral Inflow Hydrograph | BaldEagleDamBrk.p03 (Single 2D Area - Internal Dam Structure) |
| Normal Depth             | BaldEagleDamBrk.p04 (SA to 2D Area Conn - 2D Levee Structure) |

| Project | Version | Boundary Conditions | Plans |
|---------|---------|---------------------|-------|
| BaldEagleCrkMulti2D | 5.00 | - Flow Hydrograph | 1. BaldEagleDamBrk.p17: 2D to 1D No Dam |
|                     |      | - Gate Opening | 2. BaldEagleDamBrk.p18: 2D to 2D Run |
|                     |      | - Lateral Inflow Hydrograph | 3. BaldEagleDamBrk.p19: SA to 2D Dam Break Run |
|                     |      | - Uniform Lateral Inflow Hydrograph | 4. BaldEagleDamBrk.p03: Single 2D Area - Internal Dam Structure |
|                     |      | - Normal Depth | 5. BaldEagleDamBrk.p04: SA to 2D Area Conn - 2D Levee Structure |

### Need the same tables for geometry key types

Also need to open and save all example files in 6.5 using HECRASController (making a function to do this)

This can be used to help find examples if the user wants to do something that the library is currently not capable of.  

Using the RasExamples class, the user can extract these files and inspect them to find specific examples.  